src/drivers/net/intel.c

```diff
@@ -674,6 +674,19 @@ static int intel_open ( struct net_device *netdev ) {
 		  INTEL_TCTL_COLD_DEFAULT );
 	writel ( tctl, intel->regs + INTEL_TCTL );
 
+	/* The Intel I210's packet buffer size registers only reset on power up.
+	 * If an operating system changes these but then the computer recieves a
+	 * reset signal without losing power, the registers will stay the same
+	 * (but be incompatible with other register defaults), thus making the
+	 * device unable to pass traffic.
+	 */
+	if ( intel->flags & INTEL_NO_PBSIZE_RST ) {
+		writel ( INTEL_I210_RXPBSIZE_DEFAULT, intel->regs + INTEL_E1000_RXPBS );
+		writel ( INTEL_I210_TXPBSIZE_DEFAULT, intel->regs + INTEL_E1000_TXPBS );
+		writel ( INTEL_I210_DTXMXPKTSZ,
+		         intel->regs + INTEL_I210_DTXMXPKTSZ_DEFAULT );
+	}
+
```

Comment:
I think this logic probably belongs in `intel_reset()`, alongside the very similar logic for `INTEL_PBS_ERRATA`.  There is no need to reapply these register values on every device open, as far as I can tell.

Reply:
Yes, you're right; I've moved it just under the `INTEL_PBS_ERRATA` block in the latest revision